### ISSUE TITLE
Add support for GIS / spatial types

### ISSUE THREAD

**Issue by labeldevops** [redwoodjs/redwood#4718]
You should strongly consider optionally using something else than prisma, as that ORM does not support spatial types in any meaningful way, see https://github.com/prisma/prisma/issues/2789 and https://github.com/prisma/prisma/issues/1798.

Usually any web application will need spatial support at some point, so when a young team / developer tries to build something with redwood they / she will hit a very hard wall when the first geo feature will be needed and they realize that their stack has zero support for something that is quite common on the internet.

You should not build this kind of limitation into this framework. Please consider GIS and spatial types not as something exotic but make them first class citizens in your framework.

We will need many gis-aware developers to tackle climate change, so please help in not baking another framework with a "not-for-gis"-limitation.

Thanks!

**Comment by simoncrypta** [comment]
Hey @labeldevops 
it's true that Prisma don't support PostGIS yet, but it's on their public roadmap !
https://www.notion.so/Prisma-Roadmap-50766227b779464ab98899accb98295f?p=525cf298497b4872a03dc7149f09f22d
 
 Meanwhile, It's possible to write raw SQL, I know it's not as fun that writing Prisma query, but you got no limitation. :ringed_planet: 
 https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access
 
 Someone wrote how doing it in Stack Overflow : 
 https://stackoverflow.com/questions/51010009/does-prisma-supports-geospatial-queries
 
 I hope this can help you, and thank you for your suggestion ! :smiley:

**Comment by thedavidprice** [comment]
Hi @labeldevops Thanks for opening this.

> We will need many gis-aware developers to tackle climate change,

💯 This is something that's very important to us as well. And as a matter of fact you'll some some specifics in announcements to come this year. Hope this means you're working in this space, too!

> You should not build this kind of limitation into this framework. Please consider GIS and spatial types not as something exotic but make them first class citizens in your framework.

I don't really agree with the implication here that this is a) a limitation or b) an either/or. Sure, it's not an ideal experience and very much lacks first-class support, but I second what @simoncrypta wrote above as an example of a way forward.

Additionally, Redwood makes it possible to use any DB Client you want. It just ships with Prisma, but you can swap it if you'd like. (And/or do something in parallel in your app.)

## Collaborative Ways forward
We (Redwood Core Team) collaborate directly with Prisma, DB providers, Hosting providers, etc. We can't really tell each other what to put on our respective roadmaps, but we do like to help each other reach goals and solve problems that are synergistic. 

So what this could look like is if a climate-focused product, built with Redwood, was running into challenges with GIS/Spatial, then we have a thing to bring up to everyone, including an example, and ask for help and ideas to move forward. And we also have a Redwood Startup Club network that works with the Core Team, which also collaborate on challenges like this ( e.g. what features does Redwood need to support current startups?). 

Problems and obstacles normally need momentum in order to be overcome and solved. So this is how we tap into momentum.

**tl;dr**
If you know any startups focused on Climate that are looking to build with Redwood, I'd like to meet them. And I'd also like to introduce them to [PWVentures](https://prestonwernerventures.com), which is the company that sponsors Redwood.
